editor/components/media-placeholder/index.js

```diff
@@ -54,7 +55,25 @@ class MediaPlaceholder extends Component {
 	onSubmitSrc( event ) {
 		event.preventDefault();
 		if ( this.state.src ) {
-			this.props.onSelectUrl( this.state.src );
+			if ( this.props.onSelectUrl ) {
+				// TODO: In removing deprecation, ensure to simplify rendering
+				// to avoid checking for `onSelectUrl`. It also allows this
+				// function to be simplified to avoid truthiness test on
+				// `onSelectURL`, since it's required for the form invoking
+				// this function to be rendered at all.
+				deprecated( 'MediaPlaceholder `onSelectUrl` prop', {
+					alternative: '`onSelectURL` prop',
+					plugin: 'Gutenberg',
+					version: 'v3.4',
```

Comment:
`3.5` to match with others? Whatever we decide 😄

Reply:
Yes, let's go with 3.5.